### ISSUE TITLE
CSP prevents advanced rules

### ISSUE THREAD

**Issue by AdamWr** [AdguardTeam/AdGuardForSafari#917]
It's related to this issue - https://github.com/AdguardTeam/AdguardFilters/issues/163515 - the problem is that sponsored posts are not blocked on `facebook.com`, it happens with Safari extension and app for iOS.
It looks like that script execution is blocked by CSP:
```
Refused to execute a script because its hash, its nonce, or 'unsafe-inline' does not appear in the script-src directive of the Content Security Policy.
```

I'm aware that's duplicate of https://github.com/AdguardTeam/AdGuardForSafari/issues/304, there is also similar issue in browser extension repository - https://github.com/AdguardTeam/AdguardBrowserExtension/issues/1733 with a link to possibly fix - https://github.com/AdguardTeam/AdguardBrowserExtension/issues/1733#issuecomment-1396625121

As far as I understand, this is used to execute advanced rules/scripts:
https://github.com/AdguardTeam/AdGuardForSafari/blob/39ec6264aa2817387dcabed0d5c1f9cdfb98165e/AdGuard/Extension/advanced-script.js#L14-L28
unfortunately it looks like that this approach is blocked by website's Content Security Policy.
So maybe something like this:
```js
const executeScripts = scripts => {
  // Wrap with try catch 
  scripts.unshift('( function () { try {');
  scripts.push("} catch (ex) { console.error('Error executing AG js: ' + ex); } })();");
  
  const blob = new Blob([scripts.join('\r\n')], {
    type: 'text/javascript'
  });
  const url = URL.createObjectURL(blob);
  const scriptTag = document.createElement('script');
  
  const parent = document.head || document.documentElement;
  parent.appendChild(scriptTag);
  if (scriptTag.parentNode) {
    URL.revokeObjectURL(url);
  }
};
```
could be used as a workaround, it seems that it works, at least in browser's console while debugging.

**Comment by maximtop** [comment]
It will help only in the case of facebook, where blobs are not blocked by csp policies.
Seems that it is worth to be added as an additional way.
